New blog: Running Databases on Kubernetes Locally with OpenEverest (#81)
openeverest/openeverest.github.io: content/blog/running-openeverest-on-my-laptop/index.md

```bash
everestctl install --skip-db-namespace
```

you may see both commands failing like in the image below

![Screenshot 2026-05-18 at 10.11.00 PM.png](images/Screenshot_2026-05-18_at_10.11.00_PM.png)

but you do not need to worry this happens because there is no active Kubernetes cluster available yet, or the current kubeconfig is pointing to an old cluster endpoint.

> Note: OpenEverest requires a running Kubernetes cluster before installation can begin.
> 
## 2. Create a New kind Cluster

As we have seen that OpenEverest needs a Kubernetes cluster. For local testing, I like using a named kind cluster so it is obvious which context I am working with:

```bash
kind create cluster --name openeverest

```

This command will create a new Kubernetes cluster named `openeverest` using docker containers as kubernetes nodes. Once the cluster is up and running tools like `kubectl` and `everestctl` can communicate with Kubernetes API server.

We can confirm that kubectl is pointing at new cluster using following commands:

```bash
kubectl config current-context
kubectl get nodes
```

![Screenshot 2026-05-18 at 10.34.39 PM.png](images/Screenshot_2026-05-18_at_10.34.39_PM.png)

> This is how your Kubernetes cluster is ready for `everestctl` to install and work. Your output should be like the image shown above. At this stage we are ready to install everestctl.

Reviewer: small thing I noticed: `everestctl` was already installed earlier through Homebrew, so here I think we're actually installing OpenEverest into the cluster rather than the CLI itself

```suggestion
This is how your Kubernetes cluster is ready for `everestctl` to install and work. Your output should be like the image shown above. At this stage we are ready to install OpenEverest.
```

If your terminal still says that it failed to get server groups after the steps above, then you can diagnose it using the commands:

```bash
docker ps
kubectl describe node openeverest-control-plane
```

## 3. Install OpenEverest with `everestctl` CLI

The next step is installing OpenEverest itself. OpenEverest is deployed directly inside Kubernetes and relies heavily on Kubernetes native concepts such as operators, CRDs, namespaces, services, and controllers to manage database workloads.

In this setup, we will use the `everestctl` CLI in headless mode to avoid the interactive installation wizard and keep the installation process fully reproducible from the terminal.

```bash
everestctl install \
--namespaces everest \
--operator.mongodb=true \
--operator.postgresql=true \
--operator.mysql=true \
--skip-wizard
```

> This command provisions OpenEverest along with database operators for PostgreSQL, MySQL, and MongoDB. These operators are specialized Kubernetes controllers made by OpenEverest to handle various database lifecycle operations. The step sometimes can take several minutes on a laptop because it pulls images and starts multiple controllers.

Reviewer: a small wording suggestion: "made by OpenEverest" reads like OpenEverest directly builds or owns all of these operators

framing it around integration/use might make the relationship clearer

```suggestion
This command provisions OpenEverest along with database operators for PostgreSQL, MySQL, and MongoDB. These operators are specialized Kubernetes controllers integrated by OpenEverest to handle various database lifecycle operations. The step sometimes can take several minutes on a laptop because it pulls images and starts multiple controllers.
```

Author: bro they are inhouse operators btw have you even checked?

Reviewer: > bro they are inhouse operators btw have you even checked?

had another look because I wanted to double check 😄 

I was goinn off the official operator docs {README} here:
https://github.com/openeverest/openeverest-operator

it mentions that OpenEverest *"leverages Kubernetes Operators for MySQL, MongoDB, and PostgreSQL under the hood"*. 

that's why `"made by OpenEverest"` read more like ownership/implementation of all DB operators to me... unless I'm mixing up the OpenEverest operator with the underlying DB operators

Author: idts this needs to be changed.
sergey already read blog when i posted on slack

Reviewer: fair enough, was just sharing how I interpreted it while reviewing :)

Then you can see the following thing on the terminal window, this confirms that `everestctl` is installed successfully:

![Screenshot 2026-05-18 at 10.53.24 PM.png](images/Screenshot_2026-05-18_at_10.53.24_PM.png)

> Then after getting cleaned installation you can follow steps mentioned in the guide which is shown in terminal or you can just follow along with me.

Reviewer: ```suggestion
After a successful installation, you can follow steps mentioned in the guide which is shown in terminal or you can just follow along with me.
```

"cleaned installation" sounded a bit off while reading. I think "successful installation" fits better here.

Under the hood OpenEverest integrates with Percona Monitoring and Management (PMM) to monitor your database clusters. refer to the docs here: [https://openeverest.io/documentation/1.15.2/crd/crd_monitoring.html?h=monitoring](https://openeverest.io/documentation/1.15.2/crd/crd_monitoring.html?h=monitoring)

![Screenshot 2026-05-18 at 11.52.19 PM.png](images/Screenshot_2026-05-18_at_11.52.19_PM.png)

Once the database was provisioned, OpenEverest exposed a detailed component view showing the individual Kubernetes workloads backing the PostgreSQL cluster. This made it much easier to understand what was happening internally instead of treating the database as a black box.

![Screenshot 2026-05-19 at 12.01.01 AM.png](images/Screenshot_2026-05-19_at_12.01.01_AM.png)

In the screenshot above, the main PostgreSQL instance is already running successfully, while the `pgbouncer` component is still stuck in the `Pending` / `Initializing` state. After debugging the issue using `kubectl describe pod`, it turned out the cluster did not have enough available CPU resources to schedule the PgBouncer pod.

> Kubernetes scheduling is heavily resource aware, so even though the database itself was healthy, the scheduler could not place the PgBouncer pod onto the node because the requested CPU exceeded the remaining allocatable resources. This mainly happened because the cluster was running with limited resources on a laptop environment. In larger cloud or production grade Kubernetes clusters with proper node capacity, PgBouncer would typically schedule without issues and would not become a bottleneck like this.

Reviewer: ```suggestion
In the screenshot above, the main PostgreSQL instance is already running successfully, while the `pgbouncer` component is still stuck in the `Pending` / `Initializing` state. After debugging with `kubectl describe pod`, it turned out that PgBouncer could not be scheduled because the node lacked sufficient allocatable CPU resources.

This mainly happened because the cluster was running with limited resources in a laptop environment. In larger Kubernetes clusters with sufficient node capacity, PgBouncer would typically schedule without issues.
```

I felt this section explains the CPU scheduling issue in a few different ways... 

maybe we could tighten it a bit while keeping the debugging context intact, since the root cause already becomes clear after mentioning the allocatable CPU limitation.

## 7. Clean Up

Once you are done experimenting with the setup, OpenEverest can be removed cleanly using the same `everestctl` CLI that was used during installation.

```bash
everestctl uninstall
```

After uninstalling OpenEverest, the entire Kubernetes cluster itself can also be deleted using `kind`.

```bash
kind delete cluster --name openeverest
```

Since `kind` runs Kubernetes nodes as Docker containers, deleting the cluster removes the complete environment and frees the allocated resources from the system. If you ever want to start fresh again, simply recreate the cluster and repeat the installation workflow.

This is how you can run various databases locally using `everestctl` CLI with just few commands, without manually doing boring stuff. Hope you enjoyed this blog!
Happy shipping databases on Kubernetes with OpenEverest.

If you run into issues, discover interesting behavior, or want to contribute, feel free to join the OpenEverest Slack community and explore the project further. 

Also, do not forget to star the [OpenEverest GitHub repository](https://github.com/openeverest/openeverest) if you found the project interesting.

> Slack: https://cloud-native.slack.com/archives/C09RRGZL2UX

Reviewer: maybe worth linking the CNCF Slack workspace + `#openeverest-users` here instead, since archive links may not be accessible foor external readers!

```suggestion
Join the OpenEverest community on CNCF Slack (#openeverest-users):
https://communityinviter.com/apps/cloud-native/cncf
```

> Star on GitHub: [https://github.com/openeverest/openeverest?](https://github.com/openeverest/openeverest?utm_source=chatgpt.com)

Reviewer: ```suggestion
Star on GitHub: [https://github.com/openeverest/openeverest](https://github.com/openeverest/openeverest)
```

Do not publish the generated password. Blur or crop it from screenshots before sharing the post, and never share it with anyone.

## 5. Run and Open the OpenEverest UI

On kind, the cleanest access method is port-forwarding. It avoids LoadBalancer setup and keeps everything bound to localhost. Although you can check other ways in the docs. I am going with port forwarding way for this blog. Run this in a separate terminal:

```bash
kubectl port-forward svc/everest 8080:8080 -n everest-system

```

Then open the frontend web UI of OpenEverest by visiting [http://localhost:8080](http://localhost:8080)

![image.png](images/image.png)

After logging in with the password you changed, or the initial password you can see something like this:

![Screenshot 2026-05-18 at 11.14.27 PM.png](images/Screenshot_2026-05-18_at_11.14.27_PM.png)

At this point, OpenEverest is running locally on a kind cluster. The interesting part is that we did not manually install each database operator or wire custom manifests together. `everestctl` handled the platform bootstrap and namespace preparation.

## 6. Create Your First Database on OpenEverest

> From the UI, click Create Database and choose a small configuration that fits your laptop. For my first local run, I recommend PostgreSQL with the smallest available resource profile.

Author: yeah makes sense
